microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/compatibility/CorruptedJwtTest.java

```diff
@@ -0,0 +1,64 @@
+package org.jboss.eap.qe.microprofile.jwt.compatibility;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JsonWebToken;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JwtHelper;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.RsaKeyTool;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+
+/**
+ * Tests verifying that server will reject corrupted JWT. Corrupted in this case means that the construction itself is
+ * flawed (such as invalid JSON document).
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class CorruptedJwtTest {
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, CorruptedJwtTest.class.getSimpleName() + ".war")
+                .addClass(SecuredJaxRsEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(CorruptedJwtTest.class.getClassLoader().getResource("mp-config-basic.properties"),
+                        "microprofile-config.properties")
+                .addAsManifestResource(CorruptedJwtTest.class.getClassLoader().getResource("pki/key.public.pem"),
+                        "key.public.pem");
+    }
+
+    /**
+     * @tpTestDetails Supply a corrupted JWT (JSON is missing a closing bracket) to the server.
+     * @tpPassCrit Corrupted JWT is rejected and user receives 401/Unauthorized.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void supplyCorruptedJwtToServerTest(@ArquillianResource URL url) throws URISyntaxException {
+        final URL privateKeyUrl = CorruptedJwtTest.class.getClassLoader().getResource("pki/key.private.pkcs8.pem");
+        if (privateKeyUrl == null) {
+            throw new IllegalStateException("Private key wasn't found in resources!");
+        }
+
+        final RsaKeyTool keyTool = RsaKeyTool.newKeyTool(privateKeyUrl.toURI());
+
+        final JsonWebToken corruptedJsonJWT = new JwtHelper(keyTool, "issuer").generateJwtJsonCorrupted("FAKE_USER");
+
+        RestAssured.given()
+                .header("Authorization", "Bearer " + corruptedJsonJWT.getRawValue())
+                .when().get(url.toExternalForm() + "secured-endpoint")
+                .then()
+                .statusCode(401);
+    }
+
```

Comment:
unnecessary new line

Reply:
Thanks - I try to always leave an empty line between class/method brackets. Though I can surely remove the line.